packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -421,14 +441,30 @@ class Slider extends React.Component {
       [classes.vertical]: vertical,
     });
 
-    const thumbClasses = classNames(classes.thumb, commonClasses);
-
     const trackProperty = vertical ? 'height' : 'width';
     const thumbProperty = vertical ? 'top' : 'left';
     const inlineTrackBeforeStyles = { [trackProperty]: this.calculateTrackBeforeStyles(percent) };
     const inlineTrackAfterStyles = { [trackProperty]: this.calculateTrackAfterStyles(percent) };
     const inlineThumbStyles = { [thumbProperty]: `${percent}%` };
 
+    /** Start Thumb Icon Logic Here */
+    const withIcon = !!thumbIcon;
```

Comment:
Is it necessary to convert `thumbIcon` to an explicit boolean?

Reply:
can you suggest me a better approach, i'd love to do it that way. (always learning 😅 )

Reply:
If thumbIcon isn't provided, it will be undefined, so evaluate to false. You could still use !! for clarity if you prefer, but use it directly in the ternary and `classNames` rather than assigning to a variable.

Reply:
removed redundant `!!`